modules/api/src/main/java/com/intuit/wasabi/api/AssignmentsResource.java

```diff
@@ -645,11 +622,94 @@ public Response clearMetadataCache(@HeaderParam(AUTHORIZATION) @ApiParam(value =
     @Produces(APPLICATION_JSON)
     @ApiOperation(value = "Get assignments metadata cache details...")
     @Timed
-    public Response getMetadataCacheDetails(@HeaderParam(AUTHORIZATION) @ApiParam(value = EXAMPLE_AUTHORIZATION_HEADER, required = true) final String authorizationHeader) {
-        UserInfo.Username userName = authorization.getUser(authorizationHeader);
-        authorization.checkSuperAdmin(userName);
-
+    public Response getMetadataCacheDetails() {
         return httpHeader.headers().entity(assignments.metadataCacheDetails()).build();
     }
 
+    /**
+     * Convert Assignment object collection to the List of response MAP expected by the end user.
+     *
+     * @param assignments
+     * @return
+     */
+    protected List<Map<String, Object>> toBatchAssignmentResponseMap(Collection<Assignment> assignments) {
+        List<Map<String, Object>> responseList = new ArrayList<>();
+        assignments.forEach(assignment -> {
+            responseList.add(toBatchAssignmentResponseMap(assignment));
+        });
+        return responseList;
+    }
+
+    /**
+     * Convert Assignment object of batch-assignment to the response MAP expected by the end user.
+     * Batch-assignment response map can contain:
+     *  - experimentLabel
+     *  - assignment
+     *  - payload
+     *  - status
+     *
+     * @param assignment
+     *
+     * @return response map
+     *
+     */
+    protected Map<String, Object> toBatchAssignmentResponseMap(final Assignment assignment) {
```

Comment:
Cosmetic: Consider renaming method name to getResponseMapfromBatchAssignmentObject to better communicate intent.